### DISCUSSION TITLE
How can we prevent html entities for parts of the code?

### DISCUSSION THREAD

**Discussion by andrewerrico** [resend/react-email#1471] [Q&A] [answered]
Our server uses handlebars with custom helpers to inject data into our email templates. We use a custom `if_eq` helper but the quotes are converted to html entities

```
{{#if_eq value &quot;foo&quot;}}
...
{{/if_eq}}
```

How can I prevent this so the build looks like this?
```
{{#if_eq value "foo"}}
...
{{/if_eq}}
```

**Comment by gabrielmfern** [comment, marked as answer]
Currently, people generally do a `replaceAll` on the rendered HTML and use a script of their instead of our `email export`. You would just need to replace all `&quot;` with actual quotes.

---

That being said, I would recommend you have a separate Node server that renders the email templates on demand with the props so that you don't need to deal with this kind of stuff. You would send something like an HTTP POST with the props (or variables, like in handlebars) and then it would respond with the rendered HTML.

Talking with a lot of the people that go in the direction of using something like handlebars, it really is a pain, and there isn't much we can do honestly. It's just full of foot guns that you will need to think about. Here's a few that come to mind:

* How will you deal with the HTML entities placed by `react-dom`? (could be fixed by us having a custom renderer)
* How will you use our [`PreviewProps` field](https://react.email/docs/cli#how-can-i-define-props-specific-to-the-emails-preview)?
* How will you tell future people coming across the email templates that they can't use the usual React tools you would?
   * Stuff like using `map` or even just normal JavaScript syntax

**Comment by andrewerrico** [comment]
Thanks! We'll throw a preprocessor before handlebars to replace them. Hoping to move away from handlebars eventually.
